src/main/java/org/springframework/cloud/deployer/spi/cloudfoundry/AbstractCloudFoundryTaskLauncher.java

```diff
@@ -110,48 +126,69 @@ protected boolean maxConcurrentExecutionsReached() {
 
 	private Mono<TaskStatus> getStatus(String id) {
 		return requestGetTask(id)
-			.map(this::toTaskStatus)
-			.onErrorResume(isNotFoundError(), t -> {
-				logger.debug("Task for id={} does not exist", id);
-				return Mono.just(new TaskStatus(id, LaunchState.unknown, null));
-			})
-			.transform(statusRetry(id))
-			.onErrorReturn(createErrorTaskStatus(id));
+				.map(this::toTaskStatus)
+				.onErrorResume(isNotFoundError(), t -> {
+					logger.debug("Task for id={} does not exist", id);
+					return Mono.just(new TaskStatus(id, LaunchState.unknown, null));
+				})
+				.transform(statusRetry(id))
+				.onErrorReturn(createErrorTaskStatus(id));
 	}
 
 	private TaskStatus createErrorTaskStatus(String id) {
-			return new TaskStatus(id, LaunchState.error, null);
+		return new TaskStatus(id, LaunchState.error, null);
 	}
 
 	protected TaskStatus toTaskStatus(GetTaskResponse response) {
 		switch (response.getState()) {
-			case SUCCEEDED:
-				return new TaskStatus(response.getId(), LaunchState.complete, null);
-			case RUNNING:
-				return new TaskStatus(response.getId(), LaunchState.running, null);
-			case PENDING:
-				return new TaskStatus(response.getId(), LaunchState.launching, null);
-			case CANCELING:
-				return new TaskStatus(response.getId(), LaunchState.cancelled, null);
-			case FAILED:
-				return new TaskStatus(response.getId(), LaunchState.failed, null);
-			default:
-				throw new IllegalStateException(String.format("Unsupported CF task state %s", response.getState()));
+		case SUCCEEDED:
+			return new TaskStatus(response.getId(), LaunchState.complete, null);
+		case RUNNING:
+			return new TaskStatus(response.getId(), LaunchState.running, null);
+		case PENDING:
+			return new TaskStatus(response.getId(), LaunchState.launching, null);
+		case CANCELING:
+			return new TaskStatus(response.getId(), LaunchState.cancelled, null);
+		case FAILED:
+			return new TaskStatus(response.getId(), LaunchState.failed, null);
+		default:
+			throw new IllegalStateException(String.format("Unsupported CF task state %s", response.getState()));
 		}
 	}
 
 	private Mono<CancelTaskResponse> requestCancelTask(String taskId) {
 		return this.client.tasks()
-			.cancel(CancelTaskRequest.builder()
-				.taskId(taskId)
-				.build());
+				.cancel(CancelTaskRequest.builder()
+						.taskId(taskId)
+						.build());
 	}
 
 	private Mono<GetTaskResponse> requestGetTask(String taskId) {
 		return this.client.tasks()
-			.get(GetTaskRequest.builder()
-				.taskId(taskId)
-				.build());
+				.get(GetTaskRequest.builder()
+						.taskId(taskId)
+						.build());
+	}
+
+	private String organizationId() {
```

Comment:
Given the org and space properties are required to be set as the CF deployer properties and there will be a validation check during the server start up on those properties, do you think we can just use what is set on the CF deployer properties to get the `org` and `spaceId` instead of requesting CF controller for this info?

Reply:
We have the org and space name. We need the guids.